pgml-dashboard/src/lib.rs

```diff
@@ -119,6 +118,44 @@ impl Clusters {
     }
 }
 
+// Globally shared state in memory.
+
+#[derive(Debug, Clone)]
+pub struct CurrentUser {
+    pub user: Arc<Mutex<Option<models::User>>>,
+    pub visible_clusters: Arc<Mutex<Option<HashMap<String, String>>>>,
+}
+
+impl CurrentUser {
+    pub fn set_user(&self, user: models::User) {
+        *self.user.lock() = Some(user);
+    }
+
+    pub fn get_user(&self) -> Option<models::User> {
+        self.user.lock().clone()
+    }
+
+    pub fn set_visible_clusters(&self, visible_clusters: HashMap<String, String>) {
+        *self.visible_clusters.lock() = Some(visible_clusters);
+    }
+
+    pub fn get_visible_clusters(&self) -> Option<HashMap<String, String>> {
```

Comment:
I think we have a `Cluster` object we make public. We can use it here to benefit from some type safety.